IdentityCore/src/validation/MSIDAadAuthorityCache.m

```diff
@@ -74,22 +53,19 @@ + (MSIDAadAuthorityCache *)sharedInstance
     return singleton;
 }
 
-- (BOOL)processMetadata:(NSArray<NSDictionary *> *)metadata
+- (void)processMetadata:(NSArray<NSDictionary *> *)metadata
    openIdConfigEndpoint:(NSURL *)openIdConfigEndpoint
               authority:(NSURL *)authority
                 context:(id<MSIDRequestContext>)context
-                  error:(NSError **)error
+             completion:(void (^)(BOOL result, NSError *error))completion
 {
-    if (metadata != nil)
-    {
-        CHECK_CLASS_TYPE(metadata, NSArray, @"JSON metadata from authority validation is not an array");
-    }
-    
-    [self getWriteLock];
-    BOOL ret = [self processImpl:metadata authority:authority openIdConfigEndpoint:openIdConfigEndpoint context:context error:error];
-    pthread_rwlock_unlock(&_rwLock);
-    
-    return ret;
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
```

Comment:
same as above, isn't this entire function synchronous?

Reply:
This method is async.